src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/RealWorldContextTests.cs

```diff
@@ -882,5 +882,121 @@ public virtual void NullableStruct()
             Assert.Equal("Jane", person.Value.FirstName);
             Assert.Equal("Doe", person.Value.LastName);
         }
+
+        [Fact]
+        public void JsonIgnoreForCovariantProperties()
```

Comment:
I can't tell just by looking at the diff, but we should make sure that these tests are exercised both by the fast-path and metadata-based source generator.